r/R/dataset-format.R

```diff
@@ -532,6 +565,22 @@ ParquetFragmentScanOptions$create <- function(use_buffered_stream = FALSE,
   dataset___ParquetFragmentScanOptions__Make(use_buffered_stream, buffer_size, pre_buffer)
 }
 
+#' @usage NULL
+#' @format NULL
+#' @rdname FragmentScanOptions
+#' @export
+JsonFragmentScanOptions <- R6Class("JsonFragmentScanOptions", inherit = FragmentScanOptions)
```

Comment:
I've only created bindings for this here, as it mirrors the CSV implementation, but given that https://github.com/apache/arrow/issues/34940 implies that the scanner is deprecated, should I remove this? @nealrichardson

Reply:
No, it's fine, Acero needs these too, not just the old Scanner.